Dockerfile.buildx

```diff
@@ -1,14 +1,27 @@
 ARG GO_VERSION=1.12.10
-FROM golang:${GO_VERSION}-stretch
+ARG BUILDX_COMMIT=v0.3.1
 ARG BUILDX_REPO=https://github.com/docker/buildx.git
+
+FROM golang:${GO_VERSION}-stretch AS build
+ARG BUILDX_REPO
 RUN git clone "${BUILDX_REPO}" /buildx
 WORKDIR /buildx
-ARG BUILDX_COMMIT=master
+ARG BUILDX_COMMIT
 RUN git fetch origin "${BUILDX_COMMIT}":build && git checkout build
 RUN go mod download
 ARG GOOS
 ARG GOARCH
 # Keep these essentially no-op var settings for debug purposes.
 # It allows us to see what the GOOS/GOARCH that's being built for is.
-RUN GOOS=${GOOS} GOARCH=${GOARCH} go build -ldflags '-X github.com/docker/buildx/version.Version=${BUILDX_COMMIT} -X github.com/docker/buildx/version.Revision=${BUILDX_COMMIT} -X github.com/docker/buildx/version.Package=github.com/docker/buildx' -o /usr/bin/buildx ./cmd/buildx
-ENTRYPOINT ["/usr/bin/buildx"]
\ No newline at end of file
+RUN GOOS="${GOOS}" GOARCH="${GOARCH}" BUILDX_COMMIT="${BUILDX_COMMIT}" \
+ && export version="$(git describe --tags)" \
+ && export commit="$(git rev-parse --short HEAD)" \
+ && export ldflags="\
+    -X \"github.com/docker/buildx/version.Version=${version}\" \
+    -X \"github.com/docker/buildx/version.Revision=${commit}\" \
+    -X \"github.com/docker/buildx/version.Package=github.com/docker/buildx\"" \
+ && go build -ldflags "${ldflags}" -o /usr/bin/buildx ./cmd/buildx
+
+FROM golang:${GO_VERSION}-stretch
+COPY --from=build /usr/bin/buildx /usr/bin/buildx
```

Comment:
Ah! 💡 I know why it failed in this branch; before this, there was no `COPY` in the Dockerfile, and because the `COPY` was added, it started downloading the `docker.io/tonistiigi/copy:v0.1.3` image. Apparently, that one is only available for amd64 ?

Reply:
Yup; that one is only amd64, and armv7

Reply:
What's the reasoning for doing this second image?
This only produces a runnable image if the specified GOOS/GOARCH == the daemon's GOOS/GOARCH

In effect this is creating a 2nd image and taking even more space since buildx is duplicated.

Reply:
The first stage won't be tagged, so can be garbage collected, and only the second one is preserved; the final image is roughly 870MB whereas without that stage, we're tagging an image that's 1.71GB; see the commit message https://github.com/moby/moby/commit/39f1a4ab44775428e95508ef8f186a81d60dc907